api/payloads/service_offering_test.go

```diff
@@ -1,14 +1,46 @@
 package payloads_test
 
 import (
+	"net/http"
+
 	"code.cloudfoundry.org/korifi/api/payloads"
 	"code.cloudfoundry.org/korifi/api/repositories"
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
 	"github.com/onsi/gomega/types"
 )
 
-var _ = Describe("ServiceOffering", func() {
+var _ = Describe("ServiceOfferingGet", func() {
+	DescribeTable("valid query",
+		func(query string, expectedServiceOfferingGet payloads.ServiceOfferingGet) {
+			actualServiceOfferingGet, decodeErr := decodeQuery[payloads.ServiceOfferingGet](query)
+
+			Expect(decodeErr).NotTo(HaveOccurred())
+			Expect(*actualServiceOfferingGet).To(Equal(expectedServiceOfferingGet))
+		},
+		Entry("fields[service_broker]", "fields[service_broker]=guid,name", payloads.ServiceOfferingGet{IncludeBrokerFields: []string{"guid", "name"}}),
+	)
+
+	DescribeTable("invalid query",
+		func(query string, errMatcher types.GomegaMatcher) {
+			_, decodeErr := decodeQuery[payloads.ServiceOfferingGet](query)
+			Expect(decodeErr).To(errMatcher)
+		},
+		Entry("invalid service broker field", "fields[service_broker]=foo", MatchError(ContainSubstring("value must be one of: guid, name"))),
+	)
+
+	It("returns an error if a unsupported param is passed", func() {
```

Comment:
That `it` could be an entry in the `invalid query` table above. Here is a similar [example](https://github.com/cloudfoundry/korifi/blob/d0483c8b89cc8112f731f882a8dfe1485f0b6cb5/api/payloads/service_instance_test.go#L61)

Reply:
Done